### PR TITLE
feat: send message input refactor and style update

### PR DESCRIPTION
**Pull request by domw30**
### What does this do?
- The changes required include adding the Send Icon when composing a message, in place of the audio record microphone icon. 
- Refactoring for readability and structure.
- Update UI/UX.

Scroll to be added in follow up PR.

### Why are we making this change?
- as per figma and feedback

[figma1](https://www.figma.com/file/aETpyuG2AKjNcQtCjldcX0/ZERO-Messenger-%2F-Web?node-id=9680%3A494050&mode=dev)
[figma2](https://www.figma.com/file/wTOVNZonl9MmEzORwBHDuq/zDS-Web-%26-Global?node-id=9909%3A59893&mode=dev)

### How do I test this?
- spin up the messenger chat and start typing to see updates.

#### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

https://github.com/zer0-os/zOS/assets/39112648/0985af7a-2bfa-47e2-b537-4d096e8317c9

With Scroll

https://github.com/zer0-os/zOS/assets/39112648/b57f3605-6e75-4602-9d39-858634e9627f

### CODE REVIEW

File: src/components/message-input/styles.scss [hunk 1]
```diff
@@ -27,242 +27,182 @@
...
+      resize: none;
+    }
 
-    > div {
-      height: 32px !important;
-      width: 32px !important;
-      border-radius: 5px !important;
-      margin: -1px -1px 0 0 !important;
+    &__suggestions {
+      background: none !important;
```
Comment (ratik21) [inline src/components/message-input/styles.scss:185]: I am a little confused where we should use `!important` & where we should not use it. Could you please explain that..? 🤔

File: src/components/message-input/styles.scss [hunk 2]
```diff
@@ -27,242 +27,182 @@
...
-    justify-content: center;
-    height: 32px;
-    width: 32px;
+    flex-direction: column;
+    flex: 1 0 0;
+    font-size: 14px;
+    line-height: 17px;
+
+    &__wrap {
+      &__suggestions {
```
Comment (ratik21) [inline src/components/message-input/styles.scss:126]: why not use `&__wrap-suggestions` here, if it's directly a child of the same class with no additional parent css?

File: src/components/message-input/index.tsx [hunk 3]
```diff
@@ -290,78 +294,115 @@ export class MessageInput extends React.Component<Properties, State> {
   };
 
   renderInput() {
+    const hasInputValue = this.state.value?.length > 0;
+
+    console.log(this.props.isMessengerFullScreen);
```
Comment (ratik21) [inline src/components/message-input/index.tsx]: debugging log? I am not alone it seems :joy:

File: src/components/message-input/index.tsx [hunk 4]
```diff
@@ -42,6 +44,7 @@ interface State {
   isEmojisActive: boolean;
   isGiphyActive: boolean;
   isMicActive: boolean;
+  isMessengerFullScreen?: boolean;
```
Comment (ratik21) [inline src/components/message-input/index.tsx:47]: why do we have this in the State AND in the Properties?

File: src/components/message-input/giphy/styles.scss [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  &--fullscreen {
+    right: 65px;
+  }
+
+  &-grid {
+    height: 440px;
+    overflow-y: overlay;
+    margin-top: 5px;
+    .loader {
```
Comment (ratik21) [inline src/components/message-input/giphy/styles.scss:21]: so just another css question here.. why use `.classname {..}` & not `&-classname` OR `__classname` (not sure)..?

File: src/components/message-input/index.tsx [hunk 6]
```diff
@@ -1,32 +1,34 @@
...
+import classNames from 'classnames';
 import './styles.scss';
-import { MediaType } from '../../store/messages';
-import AttachmentCards from '../../platform-apps/channels/attachment-cards';
 
 export interface PublicProperties extends PublicPropertiesContainer {}
 
 export interface Properties extends PublicPropertiesContainer {
   viewMode: ViewModes;
+  isMessengerFullScreen?: boolean;
```
Comment (ratik21) [inline src/components/message-input/index.tsx:31]: I see that we're using `isFullScreen` above for the same. Maybe use that here as well (if possible)?

**Review by ratik21 — COMMENTED** [review]
mostly css doubts

File: src/components/message-input/index.tsx (hunk 4 above)
Comment (domw30) [reply src/components/message-input/index.tsx]: Good spot. Removed.

File: src/components/message-input/giphy/styles.scss (hunk 5 above)
Comment (domw30) [reply src/components/message-input/giphy/styles.scss:21]: Old styles from a previous dev. I only moved the file into a new folder here, didn't take a look at the css. Something to check out on another task :)

File: src/components/message-input/index.tsx (hunk 6 above)
Comment (domw30) [reply src/components/message-input/index.tsx:31]: I think for the `message-input`, `isMessengerFullscreen` gives it more context of what the attribute is for. Rather than assuming what exactly is `isFullScreen`

File: src/components/message-input/styles.scss (hunk 2 above)
Comment (domw30) [reply src/components/message-input/styles.scss:126]: Ha good question. I couldn't get this working that way round, very odd! For the time it was worth, I just left it as is for now.

File: src/components/message-input/index.tsx (hunk 3 above)
Comment (domw30) [reply src/components/message-input/index.tsx]: Haha, always my friend ! 

Removed.

File: src/components/message-input/styles.scss (hunk 1 above)
Comment (domw30) [reply src/components/message-input/styles.scss:185]: We use !important when we want to make sure that a certain declaration will always apply, no matter what other stylesheets or styles might also be applicable to that HTML element.

Here !important is being used to override the top and background properties of &__suggestions and the font-size of the textarea. These styles are getting applied from elsewhere in the CSS and we want to ensure these specific styles apply in these cases i.e. for suggestions class.

Use of !important should be limited though as it can make debugging difficult at times.

File: src/components/message-input/emoji-picker/emoji-picker.test.tsx [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { shallow } from 'enzyme';
 import React from 'react';
-import { ViewModes } from '../../shared-components/theme-engine';
+import { ViewModes } from '../../../shared-components/theme-engine';
```
Comment (dalefukami) [inline src/components/message-input/emoji-picker/emoji-picker.test.tsx:3]: Love these little enhancements but next time can you submit them as a separate PR either before or after the actual changes.

File: src/components/message-input/giphy/styles.scss [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  bottom: 65px;
+  right: 74px;
+  background-color: themeDeprecated.$background-color-tertiary;
+  padding: 5px;
+  border-radius: 5px;
+  border: 1px solid themeDeprecated.$input-arrow-color;
+
+  &--fullscreen {
+    right: 65px;
+  }
```
Comment (dalefukami) [inline src/components/message-input/giphy/styles.scss:15]: It feels very strange that a component at this low level needs to know whether it's fullscreen or not. Feels like we may have previously defined the positioning poorly and this is just a bandaid. Positioning

Let's leave it for now but keep this in mind for the future. Full screen mode should impact the major layout pieces but not lower level pieces directly. I think there's a missing concept here or just a funky lower level layout thing that we're not using properly.

**Review by dalefukami — COMMENTED** [review]
Just a couple of notes for future changes. I'll leave it to Ratik to approve since he's already gone through it all :)

File: src/components/message-input/styles.scss (hunk 1 above)
Comment (ratik21) [reply src/components/message-input/styles.scss:185]: ahhh, so you're basically overriding that style..got it. Thanks!

**Review by ratik21 — APPROVED** [review]
